diff --git a/etc/init.d/vz-redhat.in b/etc/init.d/vz-redhat.in
index 032ccf8..e0d5230 100755
--- a/etc/init.d/vz-redhat.in
+++ b/etc/init.d/vz-redhat.in
@@ -25,8 +25,8 @@
 
 ### BEGIN INIT INFO
 # Provides: vz
-# Required-start: $network $remote_fs $syslog
-# Required-stop:  $network $remote_fs $syslog
+# Required-Start: +iscsi pve-cluster cman clvm $network $local_fs $remote_fs $syslog
+# Required-Stop:  +iscsi pve-cluster cman clvm $network $local_fs $remote_fs $syslog
 # Should-Start:  sshd vzeventd
 # Should-Stop:  sshd vzeventd
 # Default-Start: 2 3 4 5
@@ -40,6 +40,7 @@
 
 . @SCRIPTDIR@/initd-functions
 
+PVECTL=/usr/bin/pvectl
 VZQUOTA=@SBINDIR@/vzquota
 CONFIG_DIR=@VPSCONFDIR@
 VZREBOOTDIR=@VZREBOOTDIR@
@@ -279,6 +280,13 @@ start_ves()
 	local header
 	local pid pids
 
+	sysctl -q -w net.ipv4.route.src_check=0
+
+	if [ -x ${PVECTL} ]; then
+	    # pve-manager should start the containers
+	    return;
+	fi
+
 	need_restart=""
 	# CTs that were running before a reboot
 	velist=$(ls $VZREBOOTDIR)
@@ -291,7 +299,7 @@ start_ves()
 			awk '$2 == "yes" {print $1}')
 	# Then sort by bootorder
 	test -n "$velist" && velist=$(vzlist -aH -octid -s-bootorder $velist)
-	sysctl -q -w net.ipv4.route.src_check=0
+
 	get_parallel
 	for veid in $velist; do
 		[ "${veid}" = "0" ] && continue
